src/envoy/http/jwt_auth/jwt_authenticator_test.cc

```diff
@@ -464,6 +471,34 @@ TEST_F(JwtAuthenticatorTest, TestMissedJWT) {
   auth_->Verify(headers, &mock_cb_);
 }
 
+TEST_F(JwtAuthenticatorTest, TestAllowMissingOrFailedIsTrue) {
+  // In this test, when JWT is missing, the status should still be OK
+  // because allow_missing_or_failed is true.
+  SetupConfig(kExampleConfigWithAllowMissingOrFailed);
+  EXPECT_CALL(mock_cm_, httpAsyncClientForCluster(_)).Times(0);
+  EXPECT_CALL(mock_cb_, onDone(_)).WillOnce(Invoke([](const Status &status) {
+    ASSERT_EQ(status, Status::OK);
+  }));
+
+  // Empty headers.
+  auto headers = TestHeaderMapImpl{};
+  auth_->Verify(headers, &mock_cb_);
+}
+
+TEST_F(JwtAuthenticatorTest, TestInValidJwtWhenAllowMissingOrFailedIsTrue) {
+  // In this test, when JWT is invalid, the status should still be OK
+  // because allow_missing_or_failed is true.
+  SetupConfig(kExampleConfigWithAllowMissingOrFailed);
```

Comment:
Not enough. The config have no jwt spec so no jwt validation actually run. I believe even without your flag, request will still be accepted (that a good test case to add, btw).

You need a config like the kExampleConfig, but with your flag, and setup the test similar to TestInvalidJwt (and/or all those that not returns OK). But this time, should see status is OK.

If you want thorough, can also use TestWithParams and contruct example config and expected output according to the flag value.

Reply:
Done: changed the example JSON config to contain a good JWT config and allow_missing_or_failed = true. With allow_missing_or_failed == true and an invalid JWT, the status should still be OK.

Reply:
Also changed the missing JWT test case to use a config with JWT and allow_missing_or_failed == true.

Reply:
Your kExampleConfigWithAllowMissingOrFailed doesn't have any jwt spec still. Do I missing smt?

Reply:
kExampleConfigWithAllowMissingOrFailed has been removed and the tests use
kExampleConfigWithJwtAndAllowMissingOrFailed, which has JWT spec.